pkg/cacheutil/memcached_client.go

```diff
@@ -531,6 +544,35 @@ func (c *memcachedClient) getMultiSingle(ctx context.Context, keys []string) (it
 	return items, err
 }
 
+// groupKeysByServer groups cache keys within a slice based on which server they are
+// sharded to using a memcache.ServerSelector instance. The keys are ordered so keys
+// on the same server are next to each other. Any errors encountered determining which
+// server a key should be on will result in returning keys ungrouped, in the same order
+// they were supplied in. Note that output is not guaranteed to be any particular order
+// *except* that keys will be grouped by server. The order of keys returned may change
+// from call to call.
+func (c *memcachedClient) groupKeysByServer(keys []string) []string {
```

Comment:
```suggestion
func (c *memcachedClient) sortKeysByServer(keys []string) []string {
```

Reply:
Technically we are not returning groups

Reply:
Good point! Will update to use "sort" nomenclature everywhere.